spec/index.md

```diff
@@ -4173,6 +4277,8 @@ Env = {
   time = S.time[A.canister_id];
   controllers = S.controllers[A.canister_id];
   balance = S.balances[A.canister_id];
+  reserved_balance = S.reserved_balances[A.canister_id];
+  reserved_balance_limit = S.reserved_balance_limits[A.canister_id];
```

Comment:
I suppose these values can change in Env2 (defined below) passed to post-upgrade, no?

Reply:
Yep, Env2 should use the initial reserved cycles balance since we calculate cycles to reserve at the end of the entire execution.